test/UriFactoryTest.php

```diff
@@ -0,0 +1,256 @@
+<?php
+
+declare(strict_types=1);
+
+namespace LaminasTest\Diactoros;
+
+use Laminas\Diactoros\Uri;
+use Laminas\Diactoros\UriFactory;
+use PHPUnit\Framework\TestCase;
+
+use function array_shift;
+use function sprintf;
+use function str_contains;
+use function strtolower;
+
+class UriFactoryTest extends TestCase
+{
+    public function testCreateFromSapiUsesIISUnencodedUrlValueIfPresentAndUrlWasRewritten(): void
```

Comment:
Did we have existing test coverage for `private static function marshalUriFromSapi(`? Perhaps they can just be ported here...

Reply:
Yes, the bulk of these tests were extracted from the `ServerRequestFactory` tests.